jetson/percep/pcl.cpp

```diff
@@ -186,177 +186,81 @@ void PCL::FindInterestPoints(std::vector<pcl::PointIndices> &cluster_indices,
 
 }
 
+/* --- Get Angle --- */
+//This function finds the angle off the x axis the
+//line that passes through both these points is
+//Direction of 0 is left and 1 is right
+double PCL::getAngle(int buffer, int direction, std::vector<std::vector<int>> &interest_points,
+                    shared_ptr<pcl::visualization::PCLVisualizer> viewer, std::vector<int> &obstacles){
+    double newAngle = 0;
+    //If Center Path is blocked check left until have to turn too far
+    while(newAngle > -70 && newAngle < 70){
+        
+        //Finding angle off center
+        double xDist = pt_cloud_ptr->points[obstacles.at(direction)].x;
+        double zDist = pt_cloud_ptr->points[obstacles.at(0)].z;//Length of adjacent
+        xDist += direction ? buffer+HALF_ROVER : -(buffer+HALF_ROVER); //Calculate length of opposite
+        newAngle = atan(xDist/zDist)*180/PI;//tan-1(opposite/adjacent)
+
+        //Create compareLine Functors
+        compareLine leftLine (newAngle, -HALF_ROVER);
+        compareLine rightLine (newAngle, HALF_ROVER);
+        
+        obstacles.clear();
+        
+        if(CheckPath(interest_points, viewer, obstacles, leftLine, rightLine)){
+            #if PERCEPTION_DEBUG
+            std::cout<<"!!!!!!!!!!!!!!!!!!!!!!!!!!!!!FOUND NEW PATH AT: "<<newAngle<<std::endl;
+            #endif
+            return newAngle;
+        }    
+    }
+    return 360;
+}
+
 /* --- Find Clear Path --- */
 //Returns the angle to a clear path
-double PCL::FindClearPath(std::vector<std::vector<int>> interest_points,
-                        shared_ptr<pcl::visualization::PCLVisualizer> viewer) {
+double PCL::FindClearPath(std::vector<std::vector<int>> &interest_points,
+                        shared_ptr<pcl::visualization::PCLVisualizer> viewer) {                        
+    
     #if PERCEPTION_DEBUG
         pcl::ScopeTime t ("Find Clear Path");
     #endif
 
-    int buffer = 50;
-
-    std::vector<int> obstacles; //X and Y values of the left and rightmost obstacles in path
+    // create left and right angle vlaue that default at 360 degrees
```

Comment:
`value` typo

What are left and right?

Why default to 360?

Reply:
I just had that since it was an outlying number that we wouldn't calculate. Is there a different default value that you'd suggest? We calculate angles in the range of -70 to 70